cmd/gke-gcloud-auth-plugin/cred/gcloud_token_provider.go

```diff
@@ -0,0 +1,78 @@
+package cred
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+)
+
+// gcloudConfiguration holds types unmarshaled from gcloud config in json format
+type gcloudConfiguration struct {
+	Credential struct {
+		AccessToken string    `json:"access_token"`
+		TokenExpiry time.Time `json:"token_expiry"`
+	} `json:"credential"`
+	Configuration struct {
+		Properties struct {
+			Auth struct {
+				AuthorizationTokenFile string `json:"authorization_token_file"`
+			} `json:"auth"`
+		} `json:"properties"`
+	} `json:"configuration"`
+}
+
+//
+// gcloudTokenProvider provides gcloud Oath 2.0 tokens.
+//
+type gcloudTokenProvider struct {
+	readGcloudConfigRaw func() ([]byte, error)
```

Comment:
readGcloudConfig maybe? I don't really like "raw" as a qualifier.

Reply:
Please ignore this.. Raw just means it is in bytes.. and not in a struct.